File: Programmers/C++/Lv1_소수_찾기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

// bool is_prime(int num) {
//     for (int i=2; i<=sqrt(num); i++) {
//         if (num%i == 0)
//             return false;
//     }
//     return true;
// }

int solution(int n) {
    int answer = 0;
    vector<int> arr(n+1);
    
    for (int i=2; i<arr.size(); i++)
        arr[i] = i;
   
    for (int i=2; i<=n; i++) {
        for (int j=2*i; j<=n; j+=i)
            arr[j] = 0;
    }
    
    for (int i=2; i<=n; i++) {
        if (arr[i] != 0)
            answer++;
    }
    return answer;
}

File: Programmers/Python/Level_1/Lv1_소수_찾기.py
import math

def solution(n):
    answer = 0
    for i in range(n+1):
        if prime(i) and i>1:
            answer+=1
    
    return answer

def prime(num):
    sqrt_ = int(math.sqrt(num))
    for i in range(2, sqrt_+1):
        if num%i == 0:
            return False
    return True